app-menu/qubes-session-autostart

```diff
@@ -28,18 +28,39 @@ from qubesagent.xdg import launch, find_dropins, \
     load_desktop_entry_with_dropins
 import xdg.BaseDirectory
 import os
+import re
 
 
 QUBES_XDG_CONFIG_DROPINS = '/etc/qubes/autostart'
 
 
+def check_file_existence(path: str) -> bool:
+    if os.path.isabs(path):
+        return os.path.isfile(path)
+    else:
+        return os.path.isfile(os.path.join(xdg.BaseDirectory.xdg_config_home, path))
+
+
 def entry_should_be_started(entry, environments):
     """
 
     :type entry: DesktopEntry
     """
     if entry.getHidden():
         return False
+
+    # Handle AutostartCondition keys. Not part of the official XDG specification but documented here:
+    # https://lists.freedesktop.org/archives/xdg/2007-January/007436.html
+    # Only if-exists and unless-exists keywords are supported
+    # Invalid syntax is ignored
+    if entry.get('AutostartCondition', list=True):
```

Comment:
Is this if necessary? IIUC the loop below would simply iterate 0 times if the entry is not there, no?

Reply:
Right, I thought it would return None like for dicts, but actually it returns an empty list

Reply:
Yes. But even if it would return None, you could add `|| []` and all would be fine.

Reply:
Thank you for the tip